bftengine/src/bftengine/BftExecutionEngine.cpp

```diff
@@ -0,0 +1,398 @@
+// Concord
+//
+// Copyright (c) 2022 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0
+// License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "Replica.hpp"
+#include "SharedTypes.hpp"
+#include "ControlStateManager.hpp"
+#include "performance_handler.h"
+#include <utility>
+#include "BftExecutionEngine.hpp"
+#include "messages/ClientRequestMsg.hpp"
+#include "serialize.hpp"
+#include "assertUtils.hpp"
+namespace bftEngine::impl {
+
+BftExecutionEngineBase::BftExecutionEngineBase(std::shared_ptr<IRequestsHandler> requests_handler,
```

Comment:
Hi @yontyon , this is a great enhancement from an object oriented perspective!
Now that you've encapsulated the execution logic, have you considered unit testing it?

Reply:
@teoparvanov , basically yes, but not as part of this PR.
reasons:
1. This PR is big enough already
2. As this code replaces the current execution phase, it is being tested in every Apollo test, hence, it is not urgent to have unittest for the different types of the execution engine.